### PR TITLE
fix: handle gateway.service.wsConnectionParams

### PR DESCRIPTION
**Pull request by PacoDu**
This PR fixes the `gateway.service.wsConnectionParams` that is currently defined in the documentation but not handled.

It also adds the `connection_error` message handling for the `SubscriptionClient` and the `connectionInitPayload` option.

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by PacoDu** [comment]
A little change so that `gateway.service.wsConnectionParams.connectionInitPayload` can be an async function.

File: lib/subscription-client.js [hunk 1]
```diff
@@ -36,17 +38,20 @@ class SubscriptionClient {
...
+    this.connectionInitPayload = connectionInitPayload
 
     this.connect()
   }
 
   connect () {
     this.socket = new WebSocket(this.uri, this.protocols)
 
-    this.socket.onopen = () => {
+    this.socket.onopen = async () => {
```
Comment (mcollina) [inline lib/subscription-client.js:49]: Can you add a try/catch? The `connectionInitPayload` can throw, and it will lead to an unhandledRejection

**Review by mcollina — CHANGES_REQUESTED** [review]
Please add tests for the error paths where the new function throws.

**Comment by PacoDu** [comment]
is there anything that prevent this PR from being merged ?

**Review by mcollina — APPROVED** [review]
nothing,

LGTM

**Pull request merged**